### PR TITLE
More ingress related cloudprovider methods

### PR DESCRIPTION
**Pull request by bprashanth**
@cjcullen apologies for the _third_ follow up pr adding boilerplate

### CODE REVIEW

**Comment by bprashanth** [comment]
@k8s-bot test this please

**Comment by bprashanth** [comment]
Apparently CJ is ooo. Maybe oncall can just review since it's simple enough? 
@pszczesniak (the rotation.googleplex page confusingly showed you as primary on calendar and @fgrzadkowski on the main list for the 26th).

**Comment by bprashanth** [comment]
Jeff, maybe?

**Comment by jlowdermilk** [comment]
This lgtm, but @roberthbailey may have more context.

File: pkg/cloudprovider/providers/gce/gce.go [hunk 1]
```diff
@@ -1031,6 +1041,11 @@ func (gce *GCECloud) GetGlobalForwardingRule(name string) (*compute.ForwardingRu
 	return gce.service.GlobalForwardingRules.Get(gce.projectID, name).Do()
 }
 
+// ListGlobalForwardingRules lists all GlobalForwardingRules in the project.
+func (gce *GCECloud) ListGlobalForwardingRules(name string) (*compute.ForwardingRuleList, error) {
```
Comment (roberthbailey) [inline pkg/cloudprovider/providers/gce/gce.go]: The name parameter isn't used.

File: pkg/cloudprovider/providers/gce/gce.go [hunk 2]
```diff
@@ -1130,6 +1163,11 @@ func (gce *GCECloud) DeleteInstanceGroup(name string) error {
 	return gce.waitForZoneOp(op)
 }
 
+// ListInstanceGroups lists all InstanceGroups in the project.
```
Comment (roberthbailey) [inline pkg/cloudprovider/providers/gce/gce.go]: ... lists all instance groups in the project and zone.

**Comment by roberthbailey** [comment]
LGTM modulo a couple of comments.

**Comment by bprashanth** [comment]
Done, PTAL

**Comment by roberthbailey** [comment]
LGTM.

**Pull request merged**